libyul/SideEffects.h

```diff
@@ -42,17 +56,16 @@ struct SideEffects
 	/// If true, the code can be removed without changing the semantics as long as
 	/// the whole program does not contain the msize instruction.
 	bool sideEffectFreeIfNoMSize = true;
-	/// If false, storage is guaranteed to be unchanged by the code under all
-	/// circumstances.
-	bool invalidatesStorage = false;
-	/// If false, memory is guaranteed to be unchanged by the code under all
-	/// circumstances.
-	bool invalidatesMemory = false;
+	/// If false, the execution can potentially go in an infinite loop.
+	bool deterministic = true;
```

Comment:
I'm not too happy with the name

Reply:
What do you suggest? `terminates` can be a bit misleading

Reply:
I mean we can call it by its name: `containsLoop`

Reply:
We should have the negation (so we could `&&` just like the rest.) `doesNotContainLoop` is okay though.

Reply:
What about `cannotLoop` - strictly, a function inside a recursive call chain does not contain a loop, but it could be said that it can "loop"...